src/Utils/kernel_launching.jl

```diff
@@ -241,10 +251,18 @@ the architecture `arch`.
 
     dev = Architectures.device(arch)
     loop = kernel!(dev, workgroup, worksize)
+
+    # Map out the function to use active_cells_map as an index map
+    if !isnothing(active_cells_map)
+        func = MappedFunction(loop.f, active_cells_map)
+        loop = Kernel{get_kernel_parameters(loop)..., typeof(func)}(dev, func)
+    end
+
     return loop, worksize
 end
 
-
+@inline get_kernel_parameters(::Kernel{Dev, B, W}) where {Dev, B, W} = Dev, B, W
```

Comment:
This is poorly named because this file includes a type called `KernelParameters`.

Can you define a constructor for `Kernel` that takes in `loop`, `dev`, `func`?

Reply:
We can do it, It would have to take in `{Dev, B, W}` however, because these are parametric types that do not correspond to a field of `Kernel`.